File: php/features.php
<?php
declare(strict_types=1);

// DatagovCkan SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DatagovCkanFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DatagovCkanBaseFeature();
            case "test":
                return new DatagovCkanTestFeature();
            default:
                return new DatagovCkanBaseFeature();
        }
    }
}

File: py/features.py
# DatagovCkan SDK feature factory

from feature.base_feature import DatagovCkanBaseFeature
from feature.test_feature import DatagovCkanTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DatagovCkanBaseFeature(),
        "test": lambda: DatagovCkanTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
